RHIDP-5901: readOnlyRootFilesystem option

pabel-rh opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][RHIDP#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest released and/or in-development version of RHDH, open your PR against the `main` branch and cherrypick your PR to any released branches that you want to apply your changes to. --->

**IMPORTANT: Do Not Merge - To be merged by Docs Team Only**

**Version(s):**
main, 1.5
**Issue:**
[RHIDP-5901](https://issues.redhat.com/browse/RHIDP-5901)

**Preview link:** https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-938/configuring/#readonlyrootfilesystem

rm3l commented on modules/configuring-readonlyrootfilesystem/proc-configuring-readonlyrootfilesystem-option-in-rhdh-helm-chart-deployment.adoc:
```diff
@@ -0,0 +1,15 @@
+= Configuring the readOnlyRootFilesystem option in a {product} Helm chart deployment
+
+.Procedure
+. When you are deploying {product-short} using the Helm chart, add the `readOnlyFilesystem: true` line to the `containerSecurityContext` section in your values.yaml file. For example:
```
```suggestion
. When you are deploying {product-short} using the Helm chart, add the `readOnlyRootFilesystem: true` line to the `containerSecurityContext` section in your values.yaml file. For example:
```

rm3l commented on assemblies/assembly-configuring-readonlyrootfilesystem.adoc:
```diff
@@ -0,0 +1,10 @@
+[id="{context}"]
+= Configuring readOnlyRootFilesystem in {product}
+
+The {product} deployment consists of two containers, an `initContainer` that installs the Dynamic Plugins, and a backend container which runs the application. The `initContainer` has the `readOnlyRootFilesystem` option enabled by default. To enable this option, you must have `platform administrator` permissions so that you can modify the configuration and redeploy. You can manually configure the  ...
```
```suggestion
The {product} deployment consists of two containers, an `initContainer` that installs the Dynamic Plugins, and a backend container which runs the application. The `initContainer` has the `readOnlyRootFilesystem` option enabled by default. To enable this option on the backend container, you must have `platform administrator` permissions so that you can modify the configuration and redeploy. You can manually configure the `readOnlyRootFilesystem` option on the backend container by using the following methods.
```

Otherwise, it seems like the option is not enabled on the init container, IMO.
Also, I am not sure `platform administrator` permissions are needed. I only need permission to deploy resources via Helm or permission to create/update a Custom Resource for Operator-backed deployments.

pabel-rh commented:
@rm3l , thanks for your review. I've incorporated your comments!

coreydaley commented:
/lgtm

linfraze commented on assemblies/assembly-configuring-readonlyrootfilesystem.adoc:
```diff
@@ -0,0 +1,10 @@
+[id="{context}"]
+= Configuring readOnlyRootFilesystem in {product}
+
+The {product} deployment consists of two containers, an `initContainer` that installs the Dynamic Plugins, and a backend container which runs the application. The `initContainer` has the `readOnlyRootFilesystem` option enabled by default. To enable this option on the backend container, you must either have permission to deploy resource through Helm or to create or update a CR for Operator-backed  ...
```
```suggestion
The {product} deployment consists of two containers: an `initContainer` that installs the Dynamic Plugins and a backend container that runs the application. The `initContainer` has the `readOnlyRootFilesystem` option enabled by default. To enable this option on the backend container, you must either have permission to deploy resources through Helm or to create or update a CR for Operator-backed deployments. You can manually configure the `readOnlyRootFilesystem` option on the backend container by using the following methods:

* The {product} Operator
* The {product} Helm chart
```

linfraze commented on modules/configuring-readonlyrootfilesystem/proc-configuring-readonlyrootfilesystem-option-in-rhdh-helm-chart-deployment.adoc:
```diff
@@ -0,0 +1,15 @@
+= Configuring the readOnlyRootFilesystem option in a {product} Helm chart deployment
+
+.Procedure
+. When you are deploying {product-short} using the Helm chart, add the `readOnlyRootFilesystem: true` line to the `containerSecurityContext` section in your `values.yaml` file. For example:
```
```suggestion
. In your `values.yaml` file, add `readOnlyRootFilesystem: true` to the `containerSecurityContext` section. For example:
```

- Deploying with the Helm chart is implied so we can remove that fluff
- We orient the user first (in the X file / on the X page / etc.)

linfraze commented on modules/configuring-readonlyrootfilesystem/proc-configuring-readonlyrootfilesystem-option-in-rhdh-operator-deployment.adoc:
```diff
@@ -0,0 +1,24 @@
+[id="proc-configuring-readonlyrootfilesystem-option-in-rhdh-operator-deployment"]
+= Configuring the readOnlyRootFilesystem option in a {product} Operator deployment
+
+When you are deploying {product-short} using the Operator, specify a `patch` for the `deployment` in your `{product-custom-resource-type}` custom resource (CR) that applies the `readOnlyRootFilesystem` option to the `securityContext` section in the {product-short} backend container.
```
```suggestion
When you are deploying {product-short} using the Operator, you must specify a `patch` for the `deployment` in your `{product-custom-resource-type}` custom resource (CR) that applies the `readOnlyRootFilesystem` option to the `securityContext` section in the {product-short} backend container.
```
Edited to read more like conceptual info rather than a procedural step

linfraze commented on modules/configuring-readonlyrootfilesystem/proc-configuring-readonlyrootfilesystem-option-in-rhdh-operator-deployment.adoc:
```diff
@@ -0,0 +1,24 @@
+
+.Procedure
+
+. Add the `securityContext` specification in your CR. For example:
```
```suggestion
. In your `{product-custom-resource-type}` CR, add the `securityContext` specification. For example:
```

linfraze commented on modules/configuring-readonlyrootfilesystem/proc-configuring-readonlyrootfilesystem-option-in-rhdh-operator-deployment.adoc:
```diff
@@ -0,0 +1,24 @@
+        template:
+          spec:
+            containers:
+              - name: backstage-backend
```
What is backstage-backend? Can the user name this whatever they want? If so, we have been taking this approach to clearly indicate which parts of the config file are replaceable and which are verbatim: 

`name: _<backstage_backend_container_name>_`

(Optional) You can then use callouts or definitions (formatted per OCP doc guidelines) to provide additional info and/or show an example (`backstage-backend`)

- Using replaceable values: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#command-syntax-for-replaceable-values
- Using callouts: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#callouts-in-code-blocks

rm3l replied:
`backstage-backend` is the name of the main container, as defined in the Operator default configuration. It is needed to merge the securityContext defined in the CR with what the Operator creates. The user should not change the name.

linfraze commented on modules/configuring/proc-using-the-operator-to-run-rhdh-with-your-custom-configuration.adoc:
```diff
@@ -1,7 +1,7 @@
 [id="using-the-operator-to-run-rhdh-with-your-custom-configuration"]
-= Using the {product} operator to run {product-short} with your custom configuration
+= Using the {product} Operator to run {product-short} with your custom configuration
 
-To use the {product-short} operator to run {product} with your custom configuration, create your {product-custom-resource-type} custom resource (CR) that:
+To use the {product-short} Operator to run {product} with your custom configuration, create your {product-custom-resource-type} custom resource (CR) that:
```
Thanks for the cap fix!

linfraze reviewed (commented):
Thanks for the content! A few suggestions.

pabel-rh commented:
@linfraze , I've incorporated your comments. Regarding the backstage-backend name, we cannot change this name. I've added an additional explanation in the source code. Please let me know if this is okay, @rm3l .

rm3l reviewed (approved):
/lgtm

linfraze commented on modules/configuring-readonlyrootfilesystem/proc-configuring-readonlyrootfilesystem-option-in-rhdh-operator-deployment.adoc:
```diff
@@ -0,0 +1,25 @@
+
+.Procedure
+
+. In your `{product-custom-resource-type}` CR, add the `securityContext` specification in your CR. For example:
```
```suggestion
. In your `{product-custom-resource-type}` CR, add the `securityContext` specification. For example:
```

linfraze commented on modules/configuring-readonlyrootfilesystem/proc-configuring-readonlyrootfilesystem-option-in-rhdh-operator-deployment.adoc:
```diff
@@ -0,0 +1,25 @@
+        template:
+          spec:
+            containers:
+              - name: backstage-backend <1>
+                securityContext:
+                  readOnlyRootFilesystem: true
+----
+====
+<1> Name of the main container defined in the Operator default configuration. 
```
Awesome, this is helpful to know that this is verbatim text and not a variable

linfraze commented on assemblies/assembly-configuring-readonlyrootfilesystem.adoc:
```diff
@@ -0,0 +1,10 @@
+[id="{context}"]
+= Configuring readOnlyRootFilesystem in {product}
+
+The {product} deployment consists of two containers: an `initContainer` that installs the Dynamic Plugins, and a backend container that runs the application. The `initContainer` has the `readOnlyRootFilesystem` option enabled by default. To enable this option on the backend container, you must either have permission to deploy resources through Helm or to create or update a CR for Operator-backed  ...
```
```suggestion
The {product} deployment consists of two containers: an `initContainer` that installs the Dynamic Plugins, and a backend container that runs the application. The `initContainer` has the `readOnlyRootFilesystem` option enabled by default. To enable this option on the backend container, you must either have permission to deploy resources through Helm or to create or update a CR for Operator-backed deployments. You can manually configure the `readOnlyRootFilesystem` option on the backend container by using the following methods:

* The {product} Operator
* The {product} Helm chart
```

I see that we are trying to introduce what the methods are by dropping the Operator and Helm chart modules after the colon, but I do recommend adding the bullets to close out the module with a complete statement. Also, without this little list, the user would have to get through the entire Operator module before they learned that the Helm chart is the other method.

linfraze reviewed (commented):
I wouldn't leave the _the following methods:_ dangling in the assembly, but otherwise, lgtm

pabel-rh commented:
@linfraze , thank you! Yes, that explanation makes sense. I've added the bullet points.
